### PR TITLE
Run formatter (without typedef whitespace)

### PR DESCRIPTION
**Pull request by nickrobinson251**
Same as #137 but with `whitespace_typedefs=false` (i.e. Run JuliaFormatter as a way to test out domluna/JuliaFormatter.jl#295)

```julia
julia> format(
                  "src/"; 
                  style=BlueStyle(),  # requires JuliaFormatter v0.8.2 https://github.com/domluna/JuliaFormatter.jl.git#bluestyle-1
                  always_use_return=true,
                  short_to_long_function_def=true,
                  whitespace_ops_in_indices=true,
                  remove_extra_newlines=true,
                  always_for_in=true,
                  import_to_using=true,
                  pipe_to_function_call=true,
                  whitespace_in_kwargs=false,
                  whitespace_typedefs=false,
         )
```
(fyi it's also possible to tell some bits to not be formatted https://domluna.github.io/JuliaFormatter.jl/stable/skipping_formatting/)

### CODE REVIEW

File: src/functions_math.jl [hunk 1]
```diff
@@ -58,11 +62,11 @@ with 2D type parameterization `MatrixT` and 1D `VectorT`.
 It defines the various overloads for `Base.:*` that are required.
 It should be used at the top level of a module.
 """
-macro declare_matmul(MatrixT, VectorT=nothing)
-    dim_combos = VectorT === nothing ? ((2,2),) : ((1,2), (2,1), (2,2))
+macro declare_matmul(MatrixT; VectorT=nothing)
```
Comment (nickrobinson251) [inline src/functions_math.jl:65]: oooo, this is a bug!

**Comment by nickrobinson251** [comment]
Manually fixed up a couple things (see commits) :) 

I opened this PR as a test of the Formatter, but since it works so well, i suggest we merge it 😊

File: src/cat.jl [hunk 2]
```diff
@@ -45,17 +43,18 @@ for (T, S) in [
     end
 end
 
-for (f, nf, tf, tup) in [
-    (:vcat, :_named_vcat, :_typed_vcat, ()),
-    (:hcat, :_named_hcat, :_typed_hcat, (:_,)),
-    ]
+for (f, nf, tf, tup) in
+    [(:vcat, :_named_vcat, :_typed_vcat, ()), (:hcat, :_named_hcat, :_typed_hcat, (:_,))]
```
Comment (mzgubic) [inline src/cat.jl:47]: I think this was clearer before

**Review by mzgubic — APPROVED** [review]
That's really cool, I need to look into how this formatting works :)

File: src/cat.jl (hunk 2 above)
Comment (nickrobinson251) [reply src/cat.jl:47]: 🤷

Comment (nickrobinson251) [reply src/cat.jl:47]: i agree. And this is the downside of a formatter, even though you can encode a lot of subtle rules, it's hard to write one that captures the subtley of this particular case. We could fix this one up, but it'd change again if we re-ran the formatter so i'm inclined to leave it

File: src/functions.jl [hunk 3]
```diff
@@ -174,24 +166,33 @@ for (T, S) in [
...
 
-Base.collect(x::Base.Generator{<:Iterators.ProductIterator{<:Tuple{<:NamedDimsArray,Vararg{Any}}}}) = collect_product(x)
-Base.collect(x::Base.Generator{<:Iterators.ProductIterator{<:Tuple{<:Any,<:NamedDimsArray,Vararg{Any}}}}) = collect_product(x)
-Base.collect(x::Base.Generator{<:Iterators.ProductIterator{<:Tuple{<:NamedDimsArray,<:NamedDimsArray,Vararg{Any}}}}) = collect_product(x)
+for Tup in (
+    Tuple{<:NamedDimsArray,Vararg{Any}},
+    Tuple{<:Any,<:NamedDimsArray,Vararg{Any}},
+    Tuple{<:NamedDimsArray,<:NamedDimsArray,Vararg{Any}},
+)
+    @eval function Base.collect(x::Base.Generator{<:Iterators.ProductIterator{<:Tup}})
```
Comment (nickrobinson251) [inline src/functions.jl:192]:
```suggestion
    @eval function Base.collect(x::Base.Generator{<:Iterators.ProductIterator{<:$Tup}})
```

**Comment by mzgubic** [comment]
`whitespace_in_kwargs=false` seems reasonable, is there a way of making it the default? So that it is possible to just run `format(".", style=BlueStyle())`?

**Comment by nickrobinson251** [comment]
> is there a way of making it the default? So that it is possible to just run format(".", style=BlueStyle())?

Not yet, but hopefully soon https://github.com/domluna/JuliaFormatter.jl/issues/296

**Pull request merged**